checkov/policies3d/runner.py

```diff
@@ -0,0 +1,130 @@
+            self.pbar.turn_off_progress_bar()
+
+        report = Report(self.check_type)
+
+        if not checks or not scan_reports:
+            logging.debug("No checks or reports scan.")
+            return report
+
+        self.pbar.initiate(len(checks))
+
+        reports_by_fw = {report.check_type: report for report in scan_reports}
+        for check in checks:
+            records = self.run_check(check, reports_by_fw)
+            for record in records:
+                report.add_record(record=record)
+
+        self.pbar.close()
+        return report
+
+    def run_check(self, check: Base3dPolicyCheck, reports_by_fw: dict[str, Report]) -> list[Record]:
+        records = []
+        iac_results_map = self.solve_check_iac(check, reports_by_fw)
+        cve_results_map = self.solve_check_cve(check, reports_by_fw)
+
+        for iac_resource, iac_records in iac_results_map.items():
+            for cve_resource in cve_results_map:
+                if iac_resource == cve_resource:
+                    # This means we found the combination on the same resource -> create a violation for that resource
+                    check_result = CheckResult.FAILED
+                    record = self.get_record(check, iac_records[0], check_result)
+                    record.set_guideline(check.guideline)
+                    records.append(record)
+
+        self.pbar.update()
+        return records
+
+    def solve_check_iac(self, check: Base3dPolicyCheck, reports_by_fw: dict[str, Report]) -> dict[str, list[Record]]:
+        iac_results_map: dict[str, list[Record]] = {}
+        if check.iac:
+            for fw, bc_check_ids in check.iac.items():
+                fw_report = reports_by_fw.get(fw)
+                if fw_report:
+                    fw_records = fw_report.failed_checks
+                    for record in fw_records:
+                        if record.bc_check_id in bc_check_ids:
+                            resource_id = f'{record.file_abs_path}:{record.resource}'
+                            if resource_id in iac_results_map:
+                                iac_results_map[resource_id].append(record)
+                            else:
+                                iac_results_map[resource_id] = [record]
+        return iac_results_map
+
+    def solve_check_cve(self, check: Base3dPolicyCheck, reports_by_fw: dict[str, Report]) -> dict[str, list[Record]]:
+        cve_results_map: dict[str, list[Record]] = {}
+        if check.cve:
+            cve_report = reports_by_fw.get(CheckType.SCA_IMAGE)
+            if cve_report:
+                image_results = cve_report.image_cached_results
+                risk_factor = None
+                # currently checks only for a single risk factor condition on the cves, to be extended
+                for attribute, value in check.cve.items():
+                    if attribute == CVEAttribute.RISK_FACTORS:
+                        risk_factor = value[0]
+
```

Comment:
risk factor may not be the only CVE property that we might want to check - can we have a map between the combinations attribute (`CVEAttribute`) to the sca_image report and use it?

Reply:
Good idea